datavault/__init__.py

```diff
@@ -318,158 +323,78 @@ def getTags(self, sessions, datasets):
         return sessTags, dataTags
 
 class Dataset(object):
-    def __init__(self, session, name, title=None, num=None, create=False, independents=[], dependents=[]):
+    """
+    This object basically takes care of listeners and notifications.
+    All the actual data or metadata access is proxied through to a
+    backend object.
+    """
+    def __init__(self, session, name, title=None, num=None, create=False, independents=[], dependents=[], extended=False):
         self.hub = session.hub
         self.name = name
         file_base = os.path.join(session.dir, filename_encode(name))
-        self.infofile = file_base + '.ini'
```

Comment:
the file management is moved into the backend. The CSV backend which is used to read old files still uses a separate .ini file. But the HDF5 backends that store newer datasets also put the metadata inside the hdf5 file, so everything is self-contained and there's no need for a separate .ini file.